src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisitionAddIn.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Identity.Abstractions;
+using Microsoft.Identity.Client;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// Signature for token acquisition extensions that act on the application builder.
+    /// </summary>
+    /// <param name="confidentialClientApplicationBuilder">Application builder.</param>
+    /// <param name="acquireTokenOptions">Token acquisition options.</param>
+    public delegate void BuildApplication(ConfidentialClientApplicationBuilder confidentialClientApplicationBuilder, AcquireTokenOptions acquireTokenOptions);
+
+    /// <summary>
+    /// Signature for token acquisition extensions that act on the request builder, for an app token
+    /// </summary>
+    /// <param name="builder">Builder</param>
+    /// <param name="acquireTokenOptions">Token acquisition options for the request.</param>
+    public delegate void BeforeTokenAcquisitionForApp(AcquireTokenForClientParameterBuilder builder, AcquireTokenOptions acquireTokenOptions);
```

Comment:
So this would allow you to add any extensibility APIs from MSAL? Ideally we'd have an example showing how to invoke `WithOnBeforeTokenRequest`

Reply:
yes, that's the idea. It's a draft, @bgavrilMS. A POC. It needs more work.
We could even move MSAuth 1.0 support as an add-in.

Reply:
Yes, let's do that.